build/media_source/system/js/showon.es6.js

```diff
@@ -86,6 +86,19 @@ class Showon {
               }
             });
           }
+          if (this.fields[showonData[0].field] && !this.fields[showonData[0].field].origin.length) {
+            field.classList.add('hidden');
+            field.dispatchEvent(new CustomEvent('joomla:showon-hide', {
+              bubbles: true,
+            }));
```

Comment:
```suggestion
           field.dispatchEvent(new CustomEvent('joomla:showon-hide', {
              detail: {reason: 'missingParent'}
              bubbles: true,
            }));
```

Just a suggestion.
Related comment: https://github.com/joomla/joomla-cms/pull/46488#issuecomment-3617180950
What do you think, @Fedik  @dgrammatiko  would it make sense to have a separate event here?

Reply:
I would keep it minimal. Without extra events.
If you really need to have some flag about missing parent then can do kind of:
```javascript
field.dispatchEvent(new CustomEvent('joomla:showon-hide', {
  detail: {missingParent: true}
  bubbles: true,
}));
```
Or:
```javascript
detail: {reason: 'missingParent'}
```

Reply:
Nice thanks for the suggestion.

You would then leave `field.classList.add(“hidden”);` in place?
Even if that means that in a condition such as, for example, `foo!:` or `foo!:1` are we more likely risk the undesirable side effect of data loss until the user notices it?

Otherwise, I find the event name somewhat confusing.

I don't think it will ever fit in any direction anyway. The main question is in which direction it could potentially cause more problems. 🙃

Reply:
Personalty I would not hide the field, only show the warning about misconfiguration.
But if people want we can hide.

There should not be any data lose because it just visual thing and nothing to do with field values.

Reply:
Just wasn't sure if I had understood you correctly, but that basically confirms what we discussed earlier.